packages/text-kit/src/read.ts

```diff
@@ -1,114 +1,291 @@
-interface Point {
-  line: number;
-  column: number;
+import { Point as UnistPoint, Position } from 'unist';
+
+/** Like {@link UnistPoint}, but with no offset. */
+export type Point = Omit<UnistPoint, 'offset'>;
+
+/** Like {@link UnistPoint}, but with a known source index. */
+export type SourcePoint = Required<UnistPoint>;
+
+/** Like {@link Position}, but with known indices. */
+export type SourcePosition = Position & { start: SourcePoint, end: SourcePoint };
```

Comment:
Is it possible to stick with unist `Point` and `Position` types? It seems to be a lot to maintain just for differentiating the optional offset property.

Reply:
@xiaoxinghu It's only for our return types - what we're saying is that we will accept anything with the line + column as input, but we are guaranteeing that we will provide an offset when we return points.

This doesn't restrict the API in any way - whenever something could be used where a `unist.Point` was before, it still can be.

Side note, the `Point` type here is exactly the same as the type we were using before this PR, and `SourcePosition` is the same as our old `Position`.

I'd quite like to keep `SourcePoint` and `SourcePosition`, because they make it clear that we are able to provide offset position, which means that users can use `.offset` to get the index of a point in source (guaranteed). However, I understand that the `Point` type here isn't as necessary. I'll remove the `Point` type and keep the others, lmk if this is okay!

Reply:
👍

Reply:
Done 😄